images/win/scripts/Installers/Install-Msys2.ps1

```diff
@@ -53,40 +53,38 @@ pacman.exe -Syuu  --noconfirm
 taskkill /f /fi "MODULES eq msys-2.0.dll"
 
 Write-Host "`n$dash Install msys2 packages"
-pacman.exe -S --noconfirm --needed --noprogressbar base-devel compression
+$msysPackages = (Get-ToolsetContent).MsysPackages.msys2
+pacman.exe -S --noconfirm --needed --noprogressbar $msysPackages
 taskkill /f /fi "MODULES eq msys-2.0.dll"
 
 Write-Host "`n$dash Remove p7zip/7z package due to conflicts"
 pacman.exe -R --noconfirm --noprogressbar p7zip
 
-# mingw package list
-# libxml2 can be removed from the list after the issue is fixed https://github.com/msys2/MINGW-packages/issues/8658
-$tools64 = "___clang ___clang-tools-extra ___cmake ___libxml2 ___llvm ___toolchain ___ragel"
-$tools32 = "___clang ___cmake ___libxml2 ___llvm ___toolchain ___ragel"
-
-# install mingw64 packages
-Write-Host "`n$dash Install mingw64 packages"
-$pre = "mingw-w64-x86_64-"
-pacman.exe -S --noconfirm --needed --noprogressbar $tools64.replace('___', $pre).split(' ')
-
-# install mingw32 packages
-Write-Host "`n$dash Install mingw32 packages"
-$pre = "mingw-w64-i686-"
-pacman.exe -S --noconfirm --needed --noprogressbar $tools32.replace('___', $pre).split(' ')
+# install mingw packages
+$archs = (Get-ToolsetContent).MsysPackages.mingw.arch
+foreach ($arch in $archs)
+{
+  Write-Host "Installing $arch packages"
+  $runtimePackages = ((Get-ToolsetContent).MsysPackages.mingw | Where-Object { $_.arch -eq $arch }).runtime_packages.name | ForEach-Object { "${arch}-$_" }
+  $additionalPackages = ((Get-ToolsetContent).MsysPackages.mingw | Where-Object { $_.arch -eq $arch }).additional_packages | ForEach-Object { "${arch}-$_" }
+  $packagesToInstall = $runtimePackages + $additionalPackages
+  Write-Host "The following packages will be installed: $packagesToInstall"
+  pacman.exe -S --noconfirm --needed --noprogressbar $packagesToInstall
+}
 
 # clean all packages to decrease image size
 Write-Host "`n$dash Clean packages"
 pacman.exe -Scc --noconfirm
 
-Write-Host "`n$dash Installed mingw64 packages"
-pacman.exe -Q | grep ^mingw-w64-x86_64-
-
-Write-Host "`n$dash Installed mingw32 packages"
-pacman.exe -Q | grep ^mingw-w64-i686-
-
 Write-Host "`n$dash Installed msys2 packages"
 pacman.exe -Q | grep -v ^mingw-w64-
```

Comment:
I think `^mingw-w64-` regex matches all architectures:  `^mingw-w64-x86_64- ` and `^mingw-w64-i686-`. Should we swap line 79 and 82?

Reply:
You're right, all the architectures match this pattern but It's ok since we use `grep -v` here to exclude all `MinGW` packages and display only `msys2` ones